src/util/probes.d

```diff
@@ -0,0 +1,37 @@
+/* Copyright (c) 2022 The Bitcoin Core developers
+   Distributed under the MIT software license, see the accompanying
+   file COPYING or http://www.opensource.org/licenses/mit-license.php. */
+
+/*
+
+   This file is part of the more involved process of using USDT probes on macOS
+   as described in the BUILDING CODE CONTAINING USDT PROBES section in the
+   dtrace(1) manpage https://www.unix.com/man-page/mojave/1/dtrace/
+
+   This defines the providers and specifies their probes in order to generate
+   the header file `probes.h` with the necessary tracepoints macros.
+
+*/
+
+provider net {
+        probe inbound_message(int64_t, char *, char *, char *, int64_t, unsigned char *);
+        probe outbound_message(int64_t, char *, char *, char *, int64_t, unsigned char *);
+};
+
+provider validation {
+        probe block_connected(unsigned char *, int32_t, uint64_t, int32_t, uint64_t, uint64_t);
+};
+
+provider utxocache {
+        probe flush(int64_t, uint32_t, uint64_t, uint64_t, uint8_t);
+        probe add(unsigned char *, uint32_t, uint32_t, int64_t, uint8_t);
+        probe spent(unsigned char *, uint32_t, uint32_t, int64_t, uint8_t);
+        probe uncache(unsigned char *, uint32_t, uint32_t, int64_t, uint8_t);
```

Comment:
I was going to recommend to try using `uint8_t` or similar here instead of the non-existing `bool`. I see you already did that. I think changing this is preferred over changing the actual tracepoint arguments. We might want to add a comment on this tough.

Reply:
Yes, I can't understand what I was thinking going with that overcomplicated patch solution. Changing this to `uint8_t` allows me now to generate `probes.h` during build time. 

I've still got issues with the build process on different OS resulting to failed checks, probably because of lack of experience, but I'll figure it out.